src/osgViewer/View.cpp

```diff
@@ -1130,6 +1130,23 @@ void View::removeDevice(osgGA::Device* eventSource)
     }
 }
 
+bool View::requiresUpdateSceneGraph() const
+{
+    // check if the database pager needs to update the scene
+    if (getDatabasePager()->requiresUpdateSceneGraph() || getDatabasePager()->getRequestsInProgress()) return true;
```

Comment:
getRequestsInProgress() test could be moved to DabasePager::requiresUpdateSceneGraph()